pkg/runner/operations/assert/assert.go

```diff
@@ -47,41 +48,20 @@ func (o *operation) Exec(ctx context.Context) (_err error) {
 		}
 	}
 	logger.Log(logging.Assert, logging.RunStatus, color.BoldFgCyan)
+	return o.pollForAssertion(ctx)
+}
+
+func (o *operation) pollForAssertion(ctx context.Context) error {
 	var lastErrs []error
-	err := wait.PollUntilContextCancel(ctx, internal.PollInterval, false, func(ctx context.Context) (_ bool, err error) {
-		var errs []error
-		defer func() {
-			// record last errors only if there was no real error
-			if err == nil {
+	err := wait.PollUntilContextCancel(ctx, internal.PollInterval, false, func(ctx context.Context) (bool, error) {
+		candidates, errs, err := o.fetchAndValidateCandidates(ctx)
+		if err != nil || len(errs) != 0 {
+			if !kerrors.IsTimeout(err) {
```

Comment:
is this error returned when the context is cancelled ?

Reply:
fixed